Products/CMFPlone/relationhelper.py

```diff
@@ -148,101 +151,138 @@ def restore_relations(context=None, all_relations=None):
+            target_obj = uuidToObject(item["to_uuid"])
         except KeyError:
             # brain exists but no object
             target_obj = None
+        finally:
+            if not target_obj:
+                logger.info(f'No target object found for UID {item["to_uuid"]}.')
+                # Do not continue. The source_obj needs to be updated.
+                # continue
 
-        if not source_obj:
-            logger.info(f'{item["from_uuid"]} is missing')
-            continue
-
-        if not target_obj:
-            logger.info(f'{item["to_uuid"]} is missing')
-            continue
-
+        # source_obj and target_obj are dexterity content types.
         if not IDexterityContent.providedBy(source_obj):
-            logger.info(f'{source_obj} is no dexterity content')
+            logger.info(f"{source_obj} is no dexterity content")
             continue
 
         if not IDexterityContent.providedBy(target_obj):
-            logger.info(f'{target_obj} is no dexterity content')
-            continue
+            logger.info(f"{target_obj} is no dexterity content")
+            # Do not continue. The source_obj needs to be updated.
+            # continue
 
-        from_attribute = item['from_attribute']
+        # intid for target_obj exists.
         try:
             to_id = intids.getId(target_obj)
         except KeyError as e:
-            logger.warning(f'No intid for {target_obj}')
-            continue
+            logger.warning(f"No intid for {target_obj}")
+            to_id = None
+            # continue
 
+        # Postpone linkintegrity check
+        from_attribute = item["from_attribute"]
         if from_attribute == referencedRelationship:
-            # Ignore linkintegrity for now. We'll rebuilt it at the end!
-            update_linkintegrity.add(item['from_uuid'])
+            update_linkintegrity.add(item["from_uuid"])
             continue
 
+        # Working copy relations
         if HAS_ITERATE and from_attribute == ITERATE_RELATION_NAME:
             # Iterate relations are not set as values of fields
             relation = StagingRelationValue(to_id)
             event._setRelation(source_obj, ITERATE_RELATION_NAME, relation)
             continue
 
-        field_and_schema = get_field_and_schema_for_fieldname(from_attribute, source_obj.portal_type)
+        # Relations not based on schema field
+        field_and_schema = get_field_and_schema_for_fieldname(
+            from_attribute, source_obj.portal_type
+        )
         if field_and_schema is None:
             # the from_attribute is no field
-            logger.info(f'No field. Setting relation: {item}')
+            logger.info(f"No field. Setting relation: {item}")
             event._setRelation(source_obj, from_attribute, RelationValue(to_id))
             continue
 
         field, schema = field_and_schema
-        relation = RelationValue(to_id)
+        relationvalue = to_id and RelationValue(to_id) or None
```

Comment:
In general I prefer to use `RESULT_IF_TRUE if CONDITION else RESULT_IF_FALSE` instead of `CONDITION and RESULT_IF_TRUE or RESULT_IF_FALSE`.

In most cases it does not make any difference, but it does if RESULT_IF_TRUE is itself a value which evaluates to False as a boolean.